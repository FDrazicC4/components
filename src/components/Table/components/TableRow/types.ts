import { ReactNode } from 'react';


export interface Props {


    /**
     * The data for the cells used to populate this TableRow
     */
    cells?: ReactNode[],


    /**
     * Any custom content to fill the TableRow that is different than default
     */
    children?: ReactNode,


    /**
     * The className for the TableRow
     */
    className?: string,


}
