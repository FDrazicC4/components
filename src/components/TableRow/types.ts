import React from 'react';
import { StyledParentComponent } from '@components/types';

export type TableRowBaseProps = React.HTMLAttributes<HTMLTableRowElement>;

export interface Props extends StyledParentComponent {
    component?: React.ElementType<TableRowBaseProps>,
    hover?: boolean,
    selected?: boolean,
}
