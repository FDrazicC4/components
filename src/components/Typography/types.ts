import { ReactNode } from 'react';


export type Variants = 'h1' | 'h2' | 'h3' | 'body1' | 'body2' | 'label' | 'inherit';
export type Colors = (
    'primary'
    | 'secondary'
    | 'tertiary'
    | 'success'
    | 'danger'
    | 'warning'
    | 'info'
    | 'light'
    | 'dark'
    | 'inherit'
);
export type Alignments = 'inherit' | 'left' | 'center' | 'right' | 'justify';
export type Displays = 'initial' | 'block' | 'inline' | 'inline-block';


export interface Props {

    /**
     * The content or text to be rendered. Usually just a string, but can be any valid JSX.
     */
    children: ReactNode,

    /**
     * Classes to be passed through to the component.
     */
    className?: string,

    /**
     * The overall style to be applied to the typography.
     */
    variant?: Variants,

    /**
     * The font color of the content.
     */
    color?: Colors,

    /**
     * The text alignment of the content.
     */
    align?: Alignments,

    /**
     * Can be used to override the display type of the wrapping element.
     */
    display?: Displays,

    /**
     * Adds text truncation (ellipsis) to the wrapping element.
     */
    shouldTruncate?: boolean,

    /**
     * Can be used to override the element used to wrap the content.
     */
    as?: keyof JSX.IntrinsicElements | null,

    /**
     * A tab index to be passed to the underlying element
     */
    tabIndex?: number,

}
