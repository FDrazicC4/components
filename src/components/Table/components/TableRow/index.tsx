import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';


/**
 * The `<TableRow />` component
 */
function TableRow({
    className,
    children,
    cells,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableRow',
        className,
    );


    const data = children || (cells?.map((cell: ReactElement, index: number) => (
        <td key={index}>
            {cell}
        </td>
    )));


    return (
        <tr className={classes} {...otherProps}>
            {data}
        </tr>
    );


}


export default TableRow;
