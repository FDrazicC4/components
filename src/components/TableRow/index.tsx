import React from 'react';
import clsx from 'clsx';
import { Props } from './types';
import MuiTableRow from '@material-ui/core/TableRow';

const TableRow = React.forwardRef(({ className, ...props }: Props, ref: React.Ref<any>): React.ReactElement => {


    const classes = {
        selected: 'brew-TableRow--isSelected',
        hover: 'brew-TableRow--isHovered',
    };

    const classNames = clsx(
        'brew-TableRow',
        className,
    );

    return (
        <MuiTableRow {...props} classes={classes} className={classNames} ref={ref} />
    );


});

TableRow.displayName = 'TableRow';

export default TableRow;
