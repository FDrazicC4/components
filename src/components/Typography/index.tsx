import React from 'react';
import clsx from 'clsx';
import { Typography as MuiTypography } from '@material-ui/core';
import { Props } from './types';



/**
 * The `<Typography />` component renders text using the 'Roboto' font in various styles.
 */
export const Typography = React.forwardRef(({
    variant = 'body1',
    align = 'inherit',
    display = 'initial',
    shouldTruncate = false,
    color,
    hasGutter = false,
    className,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const classNames = clsx(
        'brew-Typography',
        [`brew-Typography--display-${display}`],
        { [`brew-Typography--variant-${String(variant)}`]: variant },
        { [`brew-Typography--align-${String(align)}`]: align },
        { [`brew-Typography--color-${String(color)}`]: color },
        { 'brew-Typography--shouldTruncate': shouldTruncate },
        { 'brew-Typography--hasGutter': hasGutter },
        className,
    );


    return (
        <MuiTypography
            align={align}
            className={classNames}
            display={display}
            gutterBottom={hasGutter}
            noWrap={shouldTruncate}
            ref={ref}
            variant={variant}
            {...otherProps}
        />
    );


});


Typography.displayName = 'Typography';


export default Typography;
