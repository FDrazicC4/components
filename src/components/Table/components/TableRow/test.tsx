import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TableRow from './index';
import Typography from '../../../Typography';


describe('Table Row', () => {


    it('Renders correctly', () => {
        const content = renderer
            .create((
                <TableRow>
                    <td>Column1</td>
                    <td>Column2</td>
                    <td>Column3</td>
                </TableRow>
            ))
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <TableRow className={className}>
                <td>Column4</td>
                <td>Column5</td>
                <td>Column6</td>
            </TableRow>
        ));

        expect(content.hasClass(className)).toBe(true);
        expect(content.hasClass('brew-TableRow')).toBe(true);
    });


});

describe('Table Header Props', () => {


    it('Renders "columns" prop correctly', () => {
        const content = renderer
            .create((
                <TableRow
                    cells={['ColumnA', 'ColumnB', 'ColumnC']}
                />
            ))
            .toJSON();

        expect(content).toMatchSnapshot();

        expect(shallow((
            <TableRow
                cells={[
                    <Typography key="colA" variant="body1">ColumnA</Typography>,
                    <Typography key="colB" variant="body1">ColumnB</Typography>,
                    <Typography key="colC" variant="body1">ColumnC</Typography>,
                ]}
            />
        ))
            .find(Typography)
            .find('[variant="body1"]')
            .children()
            .contains('ColumnB')).toBe(true);
    });


});
