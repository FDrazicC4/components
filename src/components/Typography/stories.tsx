import React, { ReactElement, ReactNode } from 'react';
import { text, select, boolean } from '@storybook/addon-knobs';
import Label from '../Label';
import Typography from './index';
import { Variants, Colors, Alignments, Displays } from './types';


export default {
    component: Typography,
    parameters: {
        componentSubtitle: <Label color="success">Stable</Label>,
    },
    title: 'Components|Display/Typography',
};


export const General = (): ReactElement => {
    const align: Alignments = select('align', ['inherit', 'left', 'center', 'right', 'justify'], 'inherit');
    const children: ReactNode = text('children', 'Lorem ipsum dolor sit amet.');

    const color: Colors = select('color', [
        'initial',
        'primary',
        'secondary',
        'tertiary',
        'success',
        'warning',
        'info',
        'danger',
    ], 'initial');
    const display: Displays = select('display', ['initial', 'block', 'inline', 'inline-block'], 'initial');
    const shouldTruncate = boolean('shouldTruncate', false);
    const variant: Variants = select('variant', ['h1', 'h2', 'h3', 'body1', 'body2', 'label', 'inherit'], 'h1');

    return (
        <Typography
            align={align}
            color={color}
            display={display}
            shouldTruncate={shouldTruncate}
            variant={variant}
        >
            {children}
        </Typography>
    );
};


export const Default = (): ReactElement => (
    <div style={{ maxWidth: '100%' }}>
        <Typography>
            Default Settings - Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Typography>
    </div>
);

export const Headers = (): ReactElement => (
    <div style={{ maxWidth: '100%' }}>
        <Typography shouldTruncate variant="h1">
                H1 - Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Typography>
        <Typography shouldTruncate variant="h2">
                H2 - Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Typography>
        <Typography shouldTruncate variant="h3">
                H3 - Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Typography>
    </div>
);


export const Paragraphs = (): ReactElement => (
    <React.Fragment>
        <Typography as="p" variant="body1">
            body1 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eu ante hendrerit, imperdiet lectus in,
            egestas nulla. Donec molestie congue nulla sit amet viverra. Duis dignissim mi eget ex dignissim, id
            feugiat risus dapibus. Ut auctor imperdiet turpis, vel mollis dolor sollicitudin in.
        </Typography>
        <Typography as="p" variant="body2">
            body1 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eu ante hendrerit, imperdiet lectus in,
            egestas nulla. Donec molestie congue nulla sit amet viverra. Duis dignissim mi eget ex dignissim, id
            feugiat risus dapibus. Ut auctor imperdiet turpis, vel mollis dolor sollicitudin in.
        </Typography>
    </React.Fragment>
);


export const Truncated = (): ReactElement => (
    <Typography as="p" shouldTruncate variant="body1">
            body1 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eu ante hendrerit, imperdiet lectus in,
            egestas nulla. Donec molestie congue nulla sit amet viverra. Duis dignissim mi eget ex dignissim, id
            feugiat risus dapibus. Ut auctor imperdiet turpis, vel mollis dolor sollicitudin in.
    </Typography>
);


export const Labels = (): ReactElement => (
    <Typography as="p" shouldTruncate variant="label">
            label - Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </Typography>
);

