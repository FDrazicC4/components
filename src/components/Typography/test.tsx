import React from 'react';
import renderer from 'react-test-renderer';
import Typography from './index';


it('renders correctly', () => {
    const tree = renderer
        .create(<Typography>Foo</Typography>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders correctly with various props', () => {
    const tree = renderer
        .create((
            <React.Fragment>
                <Typography align="left" className="foo" color="primary" shouldTruncate>foo</Typography>
                <Typography align="right" color="secondary">foo</Typography>
                <Typography align="center" color="tertiary" variant="h4">foo</Typography>
            </React.Fragment>
        ))
        .toJSON();
    expect(tree).toMatchSnapshot();
});
