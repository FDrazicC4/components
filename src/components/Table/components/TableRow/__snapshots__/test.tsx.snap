// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Header Props Renders "columns" prop correctly 1`] = `
<tr
  className="brew-TableRow"
>
  <td>
    ColumnA
  </td>
  <td>
    ColumnB
  </td>
  <td>
    ColumnC
  </td>
</tr>
`;

exports[`Table Row Renders correctly 1`] = `
<tr
  className="brew-TableRow"
>
  <td>
    Column1
  </td>
  <td>
    Column2
  </td>
  <td>
    Column3
  </td>
</tr>
`;
